### PR TITLE
feat: implement payment flow in frontend with stripe

### PR DESCRIPTION
**Pull request by JeferssonCL**

### CODE REVIEW

File: apps/merchant-web/src/components/payment-status/PaymentFailed.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import PaymentStatus from "./PaymentStatus";
+import Link from "next/link";
+import styles from "@/styles/payment/StatusPayment.module.css";
+
+const FailedPayment: React.FC = () => {
```
Comment (denis-gandel) [inline apps/merchant-web/src/components/payment-status/PaymentFailed.tsx]: Don't import all of React, try to only import what is necessary

File: apps/merchant-web/src/components/payment-status/PaymentFailed.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <div>
+      <PaymentStatus
+        title="Something Went Wrong!"
+        description="We apologize for the inconvenience, but an error occurred while processing your order request."
+        imageSrc="https://res.cloudinary.com/dqowacw3b/image/upload/v1732242727/e79pqgxojgk01sdrqjye.png"
+        altText="Failure Icon"
+      >
+        <Link href="/contact-us">
+          <button className={styles["sc-btn"]}>
```
Comment (denis-gandel) [inline apps/merchant-web/src/components/payment-status/PaymentFailed.tsx:16]: This button is completely unnecessary, better to remove it

File: apps/merchant-web/src/components/payment-status/PaymentStatus.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+const PaymentStatus: React.FC<PaymentStatusProps> = ({
+                                                       title,
+                                                       description,
+                                                       imageSrc,
+                                                       altText,
+                                                       children,
+                                                     }) => {
+
+  return (
+    <div className="container xl:h-screen xl:flex xl:justify-center xl:items-center">
```
Comment (denis-gandel) [inline apps/merchant-web/src/components/payment-status/PaymentStatus.tsx]: Remember that styles must only be in .css or .module.css files

File: apps/merchant-web/src/components/payment-status/PaymentStatus.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          <p>{description}</p>
+          <p>
+            For Any Support Email:{" "}
+            <a className={styles["support-link"]} href="mailto:merchant.ecommerce.jala.support@gmail.com">
+              merchant.ecommerce.jala.support@gmail.com
+            </a>
+          </p>
+          <div className={styles["button-container"]}>
+            <Link href="/">
+              <button
```
Comment (denis-gandel) [inline apps/merchant-web/src/components/payment-status/PaymentStatus.tsx:35]: This button is completely unnecessary, better to remove it

File: apps/merchant-web/src/components/payment-status/PaymentStatus.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                className={styles["sc-btn"]}>
+                Back to HomePage
+              </button>
+            </Link>
+            {children && children}
+          </div>
+        </div>
+        <img src={imageSrc} alt={altText}/>
+      </div>
+      <style>
```
Comment (denis-gandel) [inline apps/merchant-web/src/components/payment-status/PaymentStatus.tsx]: All styles must be outside the component

File: apps/merchant-web/src/components/payment-status/PaymentStatus.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        @import url(&#39;https://fonts.googleapis.com/css2?family=Poppins:wght@800&display=swap&#39;);
+      </style>
+    </div>
+  );
+};
+
+export default PaymentStatus;
```
Comment (denis-gandel) [inline apps/merchant-web/src/components/payment-status/PaymentStatus.tsx]: Remember that it is a component and not a page

File: apps/merchant-web/src/components/payment-status/PaymentSuccessful.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
```
Comment (denis-gandel) [inline apps/merchant-web/src/components/payment-status/PaymentSuccessful.tsx]: Avoid importing React

File: apps/merchant-web/src/components/payment-status/PaymentSuccessful.tsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+        title="Payment Successful!"
+        description="Thank you for your payment! We're pleased to confirm that your transaction was successful. Your order has been processed, and you will receive an email confirmation shortly."
+        imageSrc="https://res.cloudinary.com/dqowacw3b/image/upload/v1732238312/iadthfhdbtciipddfd1l.png"
+        altText="Success Icon"
+      />
+    </div>
+  );
+};
+
+export default PaymentSuccessful;
```
Comment (denis-gandel) [inline apps/merchant-web/src/components/payment-status/PaymentSuccessful.tsx]: Remember that it is a component and not a page

File: apps/merchant-web/src/styles/payment/StatusPayment.module.css [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+  align-items: flex-start;
+  justify-content: center;
+  margin-top: 20px;
+}
+
+.sc-container img {
+  width: 700px;
+  margin-left: 200px;
+  padding-top: 40px;
+  max-width: 100%; /* Esto asegura que la imagen no se haga más pequeña de lo necesario */
```
Comment (denis-gandel) [inline apps/merchant-web/src/styles/payment/StatusPayment.module.css]: Remove this comment

File: apps/merchant-web/src/styles/payment/StatusPayment.module.css [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+.title {
+  text-align: center;
+  font-family: "Poppins", sans-serif;
+  font-size: 2.8rem;
```
Comment (denis-gandel) [inline apps/merchant-web/src/styles/payment/StatusPayment.module.css:28]: Avoid using other fonts and letter sizes that are not defined within Figma

File: apps/merchant-web/src/styles/payment/StatusPayment.module.css [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+  display: flex;
+  gap: 40px;
+  margin-top: 20px;
+  justify-content: center;
+}
+
+.sc-btn {
+  padding: 12px 20px;
+  font-family: "Poppins", sans-serif;
+  border-radius: 0.5rem;
```
Comment (denis-gandel) [inline apps/merchant-web/src/styles/payment/StatusPayment.module.css:49]: Avoid using other fonts and letter sizes that are not defined within Figma

File: apps/merchant-web/src/components/payment-status/PaymentFailed.tsx (hunk 1 above)
Comment (JeferssonCL) [reply apps/merchant-web/src/components/payment-status/PaymentFailed.tsx]: Done

File: apps/merchant-web/src/components/payment-status/PaymentFailed.tsx (hunk 2 above)
Comment (JeferssonCL) [reply apps/merchant-web/src/components/payment-status/PaymentFailed.tsx:16]: I'm following the mockup so this button must be here with the route which is initialized when a user is loggued

File: apps/merchant-web/src/components/payment-status/PaymentStatus.tsx (hunk 3 above)
Comment (JeferssonCL) [reply apps/merchant-web/src/components/payment-status/PaymentStatus.tsx]: done

File: apps/merchant-web/src/components/payment-status/PaymentStatus.tsx (hunk 4 above)
Comment (JeferssonCL) [reply apps/merchant-web/src/components/payment-status/PaymentStatus.tsx:35]: ditto

File: apps/merchant-web/src/styles/payment/StatusPayment.module.css (hunk 9 above)
Comment (JeferssonCL) [reply apps/merchant-web/src/styles/payment/StatusPayment.module.css]: done

File: apps/merchant-web/src/components/payment-status/PaymentStatus.tsx (hunk 5 above)
Comment (JeferssonCL) [reply apps/merchant-web/src/components/payment-status/PaymentStatus.tsx]: Done

File: apps/merchant-web/src/components/payment-status/PaymentSuccessful.tsx (hunk 8 above)
Comment (JeferssonCL) [reply apps/merchant-web/src/components/payment-status/PaymentSuccessful.tsx]: Done

File: apps/merchant-web/src/components/payment-status/PaymentSuccessful.tsx (hunk 7 above)
Comment (JeferssonCL) [reply apps/merchant-web/src/components/payment-status/PaymentSuccessful.tsx]: Done

File: apps/merchant-web/src/components/payment-status/PaymentStatus.tsx (hunk 6 above)
Comment (JeferssonCL) [reply apps/merchant-web/src/components/payment-status/PaymentStatus.tsx]: Done

**Review by daniel-espinoza-79 — APPROVED** [review]
Good Job

**Pull request merged**